### ISSUE TITLE
Add `--twice` (or `--ntimes=N`) flag to test reproducibility

### ISSUE THREAD

**Issue by AkihiroSuda** [docker/buildx#2053]
### Description

The `--twice` (or `--ntimes=N`) flag will run a build twice, and bail if the digests do not match.
The second run will automatically run with `--no-cache` to disable layer caches (but keep cached mounts).

The `SOURCE_DATE_EPOCH` will be set to the time of the command execution if not explicitly set.

**Comment by crazy-max** [comment]
> The `--twice` (or `--ntimes=N`) flag will run a build twice, and bail if the digests do not match.

Maybe a `--validate` flag would be better?

**Comment by AkihiroSuda** [comment]
> > The `--twice` (or `--ntimes=N`) flag will run a build twice, and bail if the digests do not match.
> 
> Maybe a `--validate` flag would be better?

`--validate` sounds confusing. `--validate-reproducibility` would be fine, but too long.

**Comment by crazy-max** [comment]
> `--validate` sounds confusing. `--validate-reproducibility` would be fine, but too long.

Let's say in the future we want another kind of validation. Maybe then `--validate` could be a slice and would take as value what we want to validate. For this case would be `--validate=repro` and for potential other validations: `--validate=repro --validate=foo ...`. cc @oyabun

**Comment by thaJeztah** [comment]
Such a flag feels more like a debugging tool; is that something a users would run normally?

**Comment by AkihiroSuda** [comment]
> Such a flag feels more like a debugging tool; is that something a users would run normally?

Eventually yes, at least on CIs.
Can be just an option for the GHA action.

**Comment by thaJeztah** [comment]
Yes, wondering indeed if having an option in GHA action would be more suitable.

Mostly concerned about overloading the (already extensive list of flags/options), especially for cases where there may already be alternatives (e.g. someone could have other tooling to verify such things).

If we _do_ think it's good to include this in the CLI, we could (should?) start looking at Cobra's features to group options. Or at least, I _THINK_ I saw they added options to group things to allow flags/options to be shown in separate sections. That way we could keep the "more commonly used" options separate from "more esoteric" options.

(But still in favour of considering _do we really need this in the cli itself_ first 🤗)

**Comment by beanscg** [comment]
Bean Labs took a public-source pass on this issue.

The narrowest useful next step looks like: Offer to identify the exact docs page and propose a minimal wording/example clarification before any patch.

Public evidence checked:
- Open public issue in `docker/buildx`; repo license `Apache-2.0`; no assignee; 7 comments.
- Labels: kind/enhancement, help wanted, area/build, status/needs-maintainer-decision, status/stale, priority/p3.
- Locator from title/labels; future commenter should ask for preferred docs/test file before drafting.

A good acceptance check would be a narrow public change that directly exercises the issue's reported boundary.

No PR opened from this pass.
